### DISCUSSION TITLE
value property of PhrasingContent is of type unknown

### DISCUSSION THREAD

**Discussion by TheOnlyTails** [remarkjs/remark#843] [Q&A] [answered]
I'm trying to match the `value` of `PhrasingContent` elements against a regex, but when I try to, my IDE warns me with the following error:
```
TS2339: Property 'match' does not exist on type 'unknown'.
```
I'm getting the type info from the `@types/mdast` package.

**Comment by wooorm** [comment, marked as answer]
I need more information. See https://github.com/remarkjs/.github/blob/main/support.md on asking quality questions.

**Reply by TheOnlyTails** [reply]
I'm `visit`ing a node of type `PhrasingContent`, and trying to take its `value` and match it against a regex to extract a specific pattern, but when I do, I get a compiler error from Typescript saying that `value`'s type is `unknown`.

Here's [my code](https://gist.github.com/099e8ba1dafbb69417fb3208d08e7673) so far.

**Reply by wooorm** [reply]
- typescript is throwing an error because your types are incorrect
- `PhrasingContent` can be many things, not just texts: https://github.com/syntax-tree/mdast#phrasingcontent.
- you are duplicating work by first finding things, and then finding and replacing things
- it’s probably simplest to first find-and-replace everything to links, add all those link nodes to an array, figure out which username/project’s exist with fetch, and then revert the links that don’t exist back to text. This also makes it easier to not send out duplicate requests
- This is probably slow due to all the network activity, perhaps whether things exist or not should be cached

**Reply by TheOnlyTails** [reply]
I see, thank you for the feedback!
